python/tank/util/login.py

```diff
@@ -35,65 +37,83 @@ def get_login_name():
 
 # note! Because the shotgun caching method can return None, to indicate that no
 # user was found, we cannot use a None value to indicate that the cache has not been
-# populated. 
+# populated.
 g_shotgun_user_cache = "unknown"
 g_shotgun_current_user_cache = "unknown"
 
+
 def get_shotgun_user(sg):
     """
-    
+
     ---- DEPRECATED ---- user get_current_user(tk) instead
-    
+
     Retrieves a shotgun user dict
     for the current user. Returns None if the user is not found in shotgun.
-    
+
     Returns the following fields:
-    
+
     * id
     * type
     * email
     * login
     * name
     * image (thumbnail)
-    
+
     This method connects to shotgun.
-    """    
-    global g_shotgun_user_cache    
-    if g_shotgun_user_cache == "unknown":        
+    """
+    global g_shotgun_user_cache
+    if g_shotgun_user_cache == "unknown":
         fields = ["id", "type", "email", "login", "name", "image"]
         local_login = get_login_name()
         g_shotgun_user_cache = sg.find_one("HumanUser", filters=[["login", "is", local_login]], fields=fields)
-    
+
     return g_shotgun_user_cache
-        
+
+
 def get_current_user(tk):
     """
-    Retrieves the current user as a dictionary of metadata values.
-    Returns None if the user is not found in shotgun.
-    
-    Returns the following fields:
-    
-    * id
-    * type
-    * email
-    * login
-    * name
-    * image url (thumbnail)
-    
-    This method connects to shotgun.
-    """    
+    Retrieves the current user as a dictionary of metadata values. Note: This method connects to
+    shotgun the first time around. The result is then cached to reduce latency.
+
+    If a user has been authenticated via a login prompt, this method will return the credentials
+    associated with that user. If Toolkit has been configured to use a script user to connect to
+    Shotgun, a core hook will be executed to established which user is associated with the current
+    session. This is usually based on the currently logged in user.
+
+    :returns: None if the user is not found in Shotgun. Otherwise, it returns a dictionary
+              with the following fields:
+                 * id
+                 * type
+                 * email
+                 * login
+                 * name
+                 * image url (thumbnail)
+    """
     global g_shotgun_current_user_cache
-    if g_shotgun_current_user_cache == "unknown":
-    
-        # call hook to get current login
-        current_login = tk.execute_core_hook(constants.CURRENT_LOGIN_HOOK_NAME)
-        if current_login is None:
-            g_shotgun_current_user_cache = None
-        
+    if g_shotgun_current_user_cache != "unknown":
+        return g_shotgun_current_user_cache
+
+    from .. import api
```

Comment:
is this to avoid cyclic imports? Just a small comment to point that out I think would be helpful!